.github/workflows/python-app.yml

```diff
@@ -12,20 +12,24 @@ jobs:
   build:
 
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
 
     steps:
     - uses: actions/checkout@v3
-    - name: Set up Python 3.10
+    - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v3
+      id: setup-python
       with:
-        python-version: "3.10"
+        python-version: ${{ matrix.python-version }}
         
     - name: cache poetry install
       uses: actions/cache@v3
       id: cache-poetry
       with:
         path: ~/.local
-        key: poetry-1.8.2-0
+        key: poetry-1.8.2-${{ steps.setup-python.outputs.python-version }}-0
```

Comment:
I'd suggest using the [built-in Poetry cache support in `actions/setup-python`](https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies) instead, which should be less brittle. That should allow you to drop the `actions/cache` steps entirely.

Reply:
Ooh, neat! Thanks for the tip.